src/pages/workspaces/workspace/analysis/file-utils.ts

```diff
@@ -71,10 +75,23 @@ export const useAnalysisFiles = (): AnalysisFileStore => {
       await Ajax(signal).AzureStorage.listNotebooks(workspaceInfo.workspaceId)
     setAnalyses(existingAnalyses)
   }) as () => Promise<void>
+
+  const create: (fullAnalysisName: any, toolLabel: ToolLabel, contents: any) => Promise<void> = _.flow(
+    // @ts-expect-error
+    withErrorReporting('Error creating analysis files'),
+    Utils.withBusyState(setPendingCreate)
+  )(async (fullAnalysisName: any, toolLabel: ToolLabel, contents: any): Promise<void> => {
+    const workspaceInfo = workspace.workspace
+    isGoogleWorkspaceInfo(workspaceInfo) ?
```

Comment:
this if/else should slot well into the provider pattern being used by a few folks to decouple the azure vs gcp differences.  I have an AnalysisProvider that might fit the bill in my pending PR